packages/components/src/menu/src/MenuContext.ts

```diff
@@ -0,0 +1,17 @@
+import { createContext } from "react";
+import type { ContextValue } from "react-aria-components";
+
+import type { MenuProps } from "./Menu.tsx";
+
+export const MenuContext = createContext<ContextValue<MenuProps<unknown>, HTMLDivElement>>({});
+
+MenuContext.displayName = "MenuContext";
+
+interface InternalMenuContextProps {
+    isSubmenu: boolean;
+    validationState?: "valid" | "invalid";
```

Comment:
we use the isInvalid props now i nstead of validation state. Is that ok here as well? 

![image](https://github.com/user-attachments/assets/348d8f69-8db9-4ff6-aa12-119afd52d528)